internal/app/machined/internal/api/reg/reg.go

```diff
@@ -507,3 +507,38 @@ func getContainerInspector(ctx context.Context, namespace string, driver common.
 		return nil, fmt.Errorf("unsupported driver %q", driver)
 	}
 }
+
+// Read implements the read API.
+func (r *Registrator) Read(in *machineapi.ReadRequest, srv machineapi.Machine_ReadServer) (err error) {
+	stat, err := os.Stat(in.Path)
+	if err != nil {
+		return err
+	}
+
+	switch mode := stat.Mode(); {
+	case mode.IsRegular():
+		f, err := os.Open(in.Path)
+		if err != nil {
+			return err
+		}
+
+		ctx, cancel := context.WithCancel(context.Background())
+
+		defer cancel()
+
+		chunker := stream.NewChunker(f)
+
+		chunkCh := chunker.Read(ctx)
```

Comment:
that is more of a question for `stream.Chunker`, but there's no way to get read errors out of it, probably worth an issue to fix later, like providing extra `errCh` back

Reply:
Yeah we can and should improve this later. We're doing a lot more streaming now.